From 492dd78abbf1cbb135543e242032ba3fd1af6f33 Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@nxp.com>
Date: Wed, 14 Mar 2018 16:08:30 +0800
Subject: [PATCH 3562/5242] MLK-17800-2: ARM64: dts: Remove the esai
 workaround for imx8qxp B0

commit  31bb5fb430b4787f872b52d7f340c883453d65a3 from
https://source.codeaurora.org/external/imx/linux-imx.git

Fixes commit 175fcc2bd433 ("MLK-15004-4: ASoC: fsl_esai: esai
workaround for imx8qxp Rev1")

The hardware issue that ESAI dma event is not connected to
EDMA correctly is fixed in B0.

Signed-off-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Reviewed-by: Viorel Suman <viorel.suman@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/freescale/fsl-imx8qxp-lpddr4-arm2.dts |    3 +--
 arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts  |    3 +--
 2 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-lpddr4-arm2.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-lpddr4-arm2.dts
index ea5a7d7..05cbbdb 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-lpddr4-arm2.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-lpddr4-arm2.dts
@@ -107,7 +107,7 @@
 };
 
 &esai0 {
-	compatible = "fsl,imx8qxp-v1-esai";
+	compatible = "fsl,imx8qm-esai";
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_esai0>;
 	assigned-clocks = <&clk IMX8QXP_ACM_ESAI0_MCLK_SEL>,
@@ -117,7 +117,6 @@
 			<&clk IMX8QXP_AUD_ESAI_0_EXTAL_IPG>;
 	assigned-clock-parents = <&clk IMX8QXP_AUD_ACM_AUD_PLL_CLK0_CLK>;
 	assigned-clock-rates = <0>, <786432000>, <49152000>, <24576000>, <49152000>;
-	dmas = <&edma2 23 0 3>, <&edma2 21 0 2>;
 	status = "okay";
 };
 
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts
index 4c96820..0b286d2 100755
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts
@@ -153,7 +153,7 @@
 };
 
 &esai0 {
-	compatible = "fsl,imx8qxp-v1-esai";
+	compatible = "fsl,imx8qm-esai";
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_esai0>;
 	assigned-clocks = <&clk IMX8QXP_ACM_ESAI0_MCLK_SEL>,
@@ -163,7 +163,6 @@
 			<&clk IMX8QXP_AUD_ESAI_0_EXTAL_IPG>;
 	assigned-clock-parents = <&clk IMX8QXP_AUD_ACM_AUD_PLL_CLK0_CLK>;
 	assigned-clock-rates = <0>, <786432000>, <49152000>, <12288000>, <49152000>;
-	dmas = <&edma2 23 0 3>, <&edma2 21 0 2>;
 	status = "okay";
 };
 
-- 
1.7.9.5

